src/components/EditCooperationPollForm.vue

```diff
@@ -0,0 +1,287 @@
+            {{ house.address?.street }}
+          </label>
+        </div>
+      </div>
+      <small v-if="v$.pollData.polledHouses.$error" id="poll_polledHouses" class="p-error">{{
+        v$.pollData.polledHouses.$errors[0].$message
+      }}</small>
+    </div>
+
+    <div class="input-section">
+      <label class="dialog-item" for="caledar-begin">Дата початку:</label>
+      <Calendar
+        id="caledar-begin"
+        v-model="pollData.creationDate"
+        :showIcon="true"
+        dateFormat="dd.mm.yy"
+        @date-select="onChangeCreationDate"
+      />
+      <small v-if="isCreationDateHelpActive" id="poll_creationDate" class="p-error">
+        Переконайтесь, що дата стоїть не раніше, ніж завтра!
+      </small>
+    </div>
+
+    <div class="input-section">
+      <label class="dialog-item" for="calendar-finish">Дата закінчення:</label>
+      <Calendar
+        id="calendar-finish"
+        v-model="pollData.completionDate"
+        :showIcon="true"
+        dateFormat="dd.mm.yy"
+        @date-select="isDisabled = false"
+        :disabled="true"
+      />
+      <small id="apartment_area_help" class="p-warning">Виставляється автоматично 15 днів з дати початку</small>
+    </div>
+  </form>
+
+  <div class="buttons-container">
+    <Button
+      label="Зберегти зміни"
+      icon="pi pi-check"
+      @click="editPoll"
+      autofocus
+      class="p-button-info"
+      :disabled="isDisabled || v$.pollData.$invalid"
+    />
+    <Button
+      label="Скасувати"
+      icon="pi pi-times"
+      @click="this.$emit('close-edit-poll')"
+      class="p-button-outlined p-button-info"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import { mapGetters } from 'vuex';
+import Button from 'primevue/button';
+import Calendar from 'primevue/calendar';
+import Checkbox from 'primevue/checkbox';
+import Textarea from 'primevue/textarea';
+import InputText from 'primevue/inputtext';
+import {
+  requiredValidator,
+  pollTitleLenghtValidator,
+  pollDescriptionLenghtValidator,
+  cyrillicLangTitleValidator,
+} from '@/utils/validators';
+import { StoreModuleEnum } from '@/store/types';
+import { HousesActionsEnum, HousesGettersEnum } from '@/store/houses/types';
+import { CooperationGettersEnum } from '@/store/cooperation/types';
+import { PollsActionEnum, PollsGettersEnum, PutPollInterface } from '@/store/polls/types';
+import { PollModel } from '@/store/polls/models/poll.model';
+import useVuelidate from '@vuelidate/core';
+import { HouseModel } from '@/shared/models/house.model';
+
+export default defineComponent({
+  name: 'EditCooperationPollForm',
+  components: {
+    Button,
+    Calendar,
+    Checkbox,
+    Textarea,
+    InputText,
+  },
+  data() {
+    return {
+      pollData: {
+        header: '',
+        description: '',
+        polledHouses: [] as Array<HouseModel>,
+        creationDate: '' as any,
+        completionDate: '' as any,
```

Comment:
are you sure its a good idea to assert string type as any?

Reply:
Initially these two variables were string type, but after certain functionality it must be changed to Date, I did try to assert String | Date, but it doesn't works. Function getDate() returns error, although getDate() works with Date object. So I was forced to set any.